crates/pessimistic-proof-test-suite/src/bin/ppgen.rs

```diff
@@ -0,0 +1,127 @@
+use std::path::PathBuf;
+
+use clap::Parser;
+use pessimistic_proof::{
+    bridge_exit::{BridgeExit, NetworkId},
+    PessimisticProofOutput,
+};
+use pessimistic_proof_test_suite::{runner::Runner, sample_data as data};
+use serde::{Deserialize, Serialize};
+use sp1_sdk::HashableKey;
+use tracing::{info, warn};
+use uuid::Uuid;
+
+/// The arguments for the pp generator.
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct PPGenArgs {
+    /// The number of bridge exits.
+    #[clap(long, default_value = "10")]
+    n_exits: usize,
+
+    /// The output directory for the generated proofs in json.
+    #[clap(long, default_value = "./data/proofs/")]
+    proof_dir: PathBuf,
+}
+
+pub fn main() {
+    sp1_sdk::utils::setup_logger();
+
+    let args = PPGenArgs::parse();
+
+    let mut state = data::sample_state_01();
+    let bridge_exits = data::sample_bridge_exits_01()
+        .take(args.n_exits)
+        .collect::<Vec<_>>();
+
+    let withdrawals = bridge_exits
+        .iter()
+        .map(|be| (be.token_info, be.amount))
+        .collect::<Vec<_>>();
+
+    let old_state = state.local_state();
+    let batch_header = state.apply_events(&[], &withdrawals);
+
+    info!(
+        "Generating the proof for {} bridge exits",
+        withdrawals.len()
+    );
+
+    let (proof, vk, new_roots) = Runner::new()
+        .generate_plonk_proof(&old_state, &batch_header)
+        .expect("proving failed");
+
+    info!("Successfully generated the plonk proof");
+
+    let vkey = vk.bytes32().to_string();
+    let fixture = PessimisticProofFixture {
+        bridge_exits,
+        pp_inputs: new_roots.into(),
+        vkey: vkey.clone(),
+        public_values: format!("0x{}", hex::encode(proof.public_values.as_slice())),
+        proof: format!("0x{}", hex::encode(proof.bytes())),
+    };
+
+    let id = Uuid::new_v4();
+    // Save the plonk proof to a json file.
+    let proof_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join(args.proof_dir);
```

Comment:
We need an argument on the CLI to disable the whole file save feature. For testnet it should just send the query to SP1, not persisting anything on disc. I'm wondering if we shouldn't integrate this directly into the agglayer node as the infra needs a docker images, metrics and logs exposition. What do you think?